ruma-federation-api/src/backfill/get_backfill/v1.rs

```diff
@@ -0,0 +1,65 @@
+//! [GET /_matrix/federation/v1/backfill/{roomId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-backfill-roomid)
+
+use std::time::SystemTime;
+
+use js_int::UInt;
+use ruma_api::ruma_api;
+use ruma_events::pdu::Pdu;
+use ruma_identifiers::{EventId, RoomId};
+
+ruma_api! {
+    metadata: {
+        description: "Request more history from another homeserver",
+        name: "get_backfill",
+        method: GET,
+        path: "/_matrix/federation/v1/backfill/:roomId",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request: {
+        /// The room ID to backfill.
+        #[ruma_api(path)]
+        pub room_id: &'a RoomId,
+
+        /// The event IDs to backfill from.
+        #[ruma_api(query)]
+        pub v: &'a [EventId],
+
+        /// The maximum number of PDUs to retrieve, including the given events.
+        #[ruma_api(query)]
+        pub limit: UInt,
+    }
+
+    response: {
+        /// The `server_name` of the homeserver sending this transaction.
+        pub origin: String,
```

Comment:
Sorry for the late feedback but I think this should be `Box<ServerName>` (from ruma-identifiers).

cc @iinuwa

Reply:
Yes, that's correct